controllers/champions.js

```diff
@@ -0,0 +1,97 @@
+const models = require('../models')
+
+const getAllChampions = async (request, response) => {
+  try {
+    const allChampions = await models.champions.findAll({
+      include: [{
+        model: models.regions,
+      },
+      {
+        model: models.roles,
+      }],
+    })
+
+    return response.send(allChampions)
+  } catch (error) {
+    return response.status(500).send('Could not find champions, please try again.')
+  }
+}
+const getChampionsByIdWithRegionAndRoles = async (request, response) => {
+  try {
+    const { id } = request.params
+
+    const matchingChampions = await models.champions.findOne({
+      include: [{
+        model: models.regions,
+      },
+      {
+        model: models.roles,
+      }],
+      where: {
+        [models.Op.or]: [
+          { id },
+          { name: { [models.Op.like]: `%${id.toLowerCase()}%` } }
+        ]
+      }
+    })
+
+    return matchingChampions
+      ? response.send(matchingChampions)
+      : response.sendStatus(404)
+  } catch (error) {
+    return response.status(500).send('Could not retrieve the champion, please try again.')
+  }
+}
+
+const postChampion = async (request, response) => {
+  try {
+    const { name, region, role } = request.body
+
+    if (!name || !region || !role) {
+      return response.status(400).send('Please fill out all required fields')
+    }
+
+    const foundRegion = await models.regions.findOne({ where: { name: region } })
+
+    const foundRole = await models.roles.findOne({ where: { role } })
+    const newChampion = await models.champions.findOrCreate({
+      where: { name },
+      defaults: { regionId: foundRegion.id }
+    })
+
+    await models.championRoles.findOrCreate({
+      where: {
+        championId: newChampion.id,
+        roleId: foundRole.id
+      }
+    })
+
+
+    return response.status(201).send(newChampion)
```

Comment:
You should have some error handling in here. What if you can't find the region based on the name I gave you? Then this route will throw an error rather than letting me know that is what the problem is.